ext/cl_khr_layers.asciidoc

```diff
@@ -0,0 +1,182 @@
+// Copyright 2017-2020 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+[[cl_khr_layers-opencl]]
+== Installable OpenCL Loader Layers
+
+[[cl_khr_layers-overview]]
+=== Overview
+
+This section describes *cl_khr_layers* loader extension which defines
+a simple mechanism through which the Khronos OpenCL installable client
+driver loader (ICD Loader) may expose load multiple separate installable
+interception layers (Layer ICDs) for OpenCL.
+
+=== General information
+
+==== Version history
+
+[cols="1,1,3",options="header",]
+|====
+| *Date*     | *Version* | *Description*
+| 2020-11-04 | 1.0.0     | First assigned version.
+|====
+
+[[cl_khr_layers-new-procedures-and-functions]]
+=== New Procedures and Functions
+
+A layer needs to implement and expose those two new entry points in
+a shared library. If one or both of those are missing, the loader will
+discard the layer.
+
+[source,c]
+----
+cl_int clGetLayerInfo(cl_layer_info  param_name,
+                      size_t         param_value_size,
+                      void          *param_value,
+                      size_t        *param_value_size_ret);
+
+cl_int clInitLayer(cl_uint                 num_entries,
```

Comment:
Should these definitions include `CL_API_ENTRY` and `CL_API_CALL`?  I'm not entirely sure if it is required, the pending header PR sets them there.

These entry points, however, differ from all others because end users will write implementations of them rather than just implementers.

Reply:
I thought that the documentation part had to omit them. I could be wrong though.

Reply:
I checked other extension file defining API entry points and the `CL_API_ENTRY` `CL_API_CALL` were omitted, even if they defined entry points that had to be implemented (`clIcdGetPlatformIDsKHR`). So I would leave them out for now. We can always change it should we deem it better.